### PR TITLE
[Feature] chunk actor logprob computation for memory saving

### PR DESCRIPTION
**Pull request by tina-wen**
### Description
This PR adds chunking along the seq_len dimension when computing actor_logprob. 
The computation now processes in chunks iteratively, significantly reducing peak memory usage.

### Key Changes
- Add chunking logic for seq_len dimension in actor_logprob computation
- Process logprob calculation in chunks to trade compute for memory
- Configurable via chunk_size in WorkerConfig.loss_cfg (BaseRLLossConfig, default: 1024)

### CODE REVIEW

File: xtuner/v1/rl/base/worker.py [hunk 1]
```diff
@@ -373,8 +375,12 @@ def compute_actor_logprobs(
         self._engine._maybe_precompute_float8_dynamic_scale_for_fsdp()
         old_logprobs_list: list[torch.Tensor] = []
         for seq_ctx, shifted_labels in zip(seq_ctx_list, shifted_labels_list):
-            output = self._engine.forward_only(seq_ctx=seq_ctx)
-            old_logprobs = gather_logprobs(output["logits"], shifted_labels)
+            if self.logprob_chunk_size is not None:
+                loss_ctx = LogProbContext(chunk_size=self.logprob_chunk_size, shifted_labels=shifted_labels)
```
Comment (HAOCHENYE) [inline xtuner/v1/rl/base/worker.py]: Please add `LogProbContextConfig` like `CELossConfig`, loss_ctx should be built by `loss_cfg.build`

**Comment by HAOCHENYE** [comment]
@claude review

File: xtuner/v1/rl/base/worker.py [hunk 2]
```diff
@@ -373,8 +375,15 @@ def compute_actor_logprobs(
         self._engine._maybe_precompute_float8_dynamic_scale_for_fsdp()
         old_logprobs_list: list[torch.Tensor] = []
         for seq_ctx, shifted_labels in zip(seq_ctx_list, shifted_labels_list):
-            output = self._engine.forward_only(seq_ctx=seq_ctx)
-            old_logprobs = gather_logprobs(output["logits"], shifted_labels)
+            if self.logprob_chunk_size is not None:
+                mode = "chunk"
+            else:
+                mode = "eager"
+            loss_ctx = LogProbConfig(chunk_size=self.logprob_chunk_size, mode=mode).build(
```
Comment (YanhuiDua) [inline xtuner/v1/rl/base/worker.py]: build方法是不是应该init时调用呢？

File: xtuner/v1/rl/base/worker.py [hunk 3]
```diff
@@ -373,8 +375,12 @@ def compute_actor_logprobs(
...
         old_logprobs_list: list[torch.Tensor] = []
         for seq_ctx, shifted_labels in zip(seq_ctx_list, shifted_labels_list):
-            output = self._engine.forward_only(seq_ctx=seq_ctx)
-            old_logprobs = gather_logprobs(output["logits"], shifted_labels)
+            if self.logprob_chunk_size is not None:
+                loss_ctx = LogProbContext(chunk_size=self.logprob_chunk_size, shifted_labels=shifted_labels)
+            else:
+                loss_ctx = None
+            output = self._engine.forward_only(seq_ctx=seq_ctx, loss_ctx=loss_ctx)
+            old_logprobs = output["loss"]
```
[the remark this answers is not in the document]
Comment (YanhuiDua) [reply xtuner/v1/rl/base/worker.py]: eager_mode时会在forward里面算gather_logprobs(output["logits"], shifted_labels)吗？

File: xtuner/v1/loss/rl_loss.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+DEVICE = get_device()
+
+
+class LogProbConfig(BaseLossConfig):
+    @property
+    def loss_ctx_cls(self) -> type["LogProbContext"]:
+        return LogProbContext
+
+    def build(self, shifted_labels: torch.Tensor, sp_mesh: DeviceMesh | None = None) -> "LogProbContext":
```
Comment (HAOCHENYE) [inline xtuner/v1/loss/rl_loss.py:21]:
```suggestion
    def build(self, shifted_labels: torch.Tensor, sp_mesh: DeviceMesh | None = None) -> Self:
```

File: xtuner/v1/loss/rl_loss.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+class LogProbKwargs(BaseLossKwargs):
+    shifted_labels: torch.Tensor
+
+
+class LogProbContext(BaseLossContext):
+    loss_cfg: LogProbConfig
+    loss_kwargs: LogProbKwargs
+
+    @classmethod
+    def build_batches_loss_kwargs(
```
Comment (HAOCHENYE) [inline xtuner/v1/loss/rl_loss.py]: This interface has been deprecated? Will it be still used?

File: xtuner/v1/rl/base/worker.py (hunk 2 above)
Comment (HAOCHENYE) [reply xtuner/v1/rl/base/worker.py]: The initialization of `loss_cfg` should be done in `__init__`, while `build` should be called here since `shifted_labels` should be passed

File: xtuner/v1/rl/base/worker.py (hunk 1 above)
Comment (tina-wen) [reply xtuner/v1/rl/base/worker.py]: Done. Added LogProbContextConfig and updated the code to build loss_ctx using loss_cfg.build().

File: xtuner/v1/rl/base/worker.py (hunk 2 above)
Comment (tina-wen) [reply xtuner/v1/rl/base/worker.py]: Fixed. Initialized self.logprob_cfg in __init__ and called build with shifted_labels here.

**Review by HAOCHENYE — APPROVED** [review]

**Review by YanhuiDua — APPROVED** [review]
